apps/debug/sysdbgapp_init.c

```diff
@@ -0,0 +1,228 @@
+	int ret = -1;
+	char *buffer = NULL;
+	if (g_sysdbg_handle < 0) {
+		/* Open the /dev/sysdbg device and read device data */
+		if (sysdbg_dev_open() < 0) {
+			return;
+		}
+	}
+	/* Passing null buffer as sysdbg device is not using this buffer to pass
+	 * data to user space. It merely prints the output using lldbg */
+	ret = read(g_sysdbg_handle, buffer, strlen(buffer));
+	if (ret < 0) {
+		printf("sysdbg: Couldn't read the device\n");
+	}
+}
+
+/****************************************************************************
+ * Name: sysdbg_dev_close
+ *
+ * Description:
+ *   This function closes sysdbg device and initialize the device handle to -1
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *  None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+static void sysdbg_dev_close(void)
+{
+	int ret = -1;
+	if (g_sysdbg_handle < 0) {
+		printf("sysdbg:  /dev/sysdbg device is not yet open\n");
+	} else {
+		ret = close(g_sysdbg_handle);
+		if (ret < 0) {
+			printf("sysdbg: failed to close /dev/sysdbg\n");
+		} else {
+			g_sysdbg_handle = -1;
+		}
+	}
+}
+
+/****************************************************************************
+ * Name:  show_usage
+ *
+ * Description:
+ *   This function prints the usage of sysdbg tash command
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+static void show_usage(FAR const char *progname)
+{
+	printf("USAGE: \n");
+	printf("kindly provide 2 arguments\n");
+	printf("For help, %s help\n", progname);
+}
+
+/****************************************************************************
+ * Name:  sysdbgapp_main
+ *
+ * Description:
+ *   This function handles the user inputs and writes it to sysdbg device
+ *
+ * Inputs:
+ *   argc - Number of arguments passed
+ *   argv - Array of character pointers
+ *
+ * Return Value:
+ *   int - zero indicates successful execution
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+int sysdbgapp_main(int argc, char **args)
+{
+	int i, nbytes;
+	char arg_str[128];
+	if (argc < 2) {
+		show_usage(args[0]);
+	} else if (strcmp("read", args[1]) == 0) {
+		sysdbg_dev_read();
+	} else {
+		/* Open system debug device created in kernel/debug/sysdbg.c */
+		if (sysdbg_dev_open() < 0) {
+			return 0;
+		}
+
+		if (strcmp("max_task_count", args[1]) == 0 || strcmp("max_irq_count", args[1]) == 0 || strcmp("max_sem_count", args[1]) == 0) {
+			i = 1;
+			memset((void *)arg_str, 0, strlen(arg_str));
+			while (argc > 1) {
+				if (i == 1) {
+					sprintf(arg_str, "%s", args[i]);
+				} else {
+					sprintf(arg_str + strlen(arg_str), " %s", args[i]);
+				}
+				argc--;
+				i++;
+			}
+			nbytes = write(g_sysdbg_handle, arg_str, strlen(arg_str));
+		} else {
+			nbytes = write(g_sysdbg_handle, args[1], strlen(args[1]));
```

Comment:
You split locations of parsing argurment. How about do it in same logic?

Reply:
handled in single logic

Reply:
No.
```
+	dbg("USAGE:\n");
 +	dbg("sysdbg enable  /* Allocates memory and enables logging */\n"); <- os
 +	dbg("sysdbg disable /* Frees the memory and disables logging */\n"); <- os
 +	dbg("sysdbg read   /* Reads /dev/sysdbg to view debug information*/\n"); <- apps
 +	dbg("sysdbg max_task_count <count> /* Enter max_task_count */\n"); <- apps
 +	dbg("sysdbg max_irq_count <count> /* Enter max_irq_count */\n"); <- apps
 +	dbg("sysdbg max_sem_count <count> /* Enter max_sem_count */\n");<- apps
 +	dbg("sysdbg data_abort /* Simulate data abort */\n"); <- os
 +	dbg("sysdbg prefetch_abort /* Simulate Prefetch abort */\n"); <- os
 +	dbg("sysdbg usr_assert /* Simulate User Assert */\n"); <- os
 +	dbg("sysdbg undef_abort  /* Simulate Undefined abort */\n"); <- os
 +	dbg("sysdbg dump_stack /* Dumps the current task stack */\n"); <- os
 +	dbg("sysdbg dump_allstack /* Dumps the stack of all tasks */\n"); <- os
```

Reply:
Sorry, I mis understood your comment earlier. I will handle it